phoenix-core/src/main/java/org/apache/phoenix/schema/types/PUnsignedLong.java

```diff
@@ -95,6 +95,11 @@ public Object toObject(byte[] b, int o, int l, PDataType actualType, SortOrder s
   }
 
   @Override
+    public boolean isCastableTo(PDataType targetType) {
```

Comment:
You'd never define a column with a type of integer and cast it to a DATE/TIME. The "constant to DATE/TIME/TIMESTAMP" isn't a real use case. Given there's a workaround already even for unsigned timestamp (casting it to a BIGINT first), I'm somewhat reluctant to do anything.